File: test_dct.py
import numpy as np
import scipy.fft as fft

x = np.array([1.0, 2.0, 3.0, 4.0])
print("dct1:", fft.dct(x, type=1))
print("dct2:", fft.dct(x, type=2))
print("dct3:", fft.dct(x, type=3))
print("dct4:", fft.dct(x, type=4))
print("idct2:", fft.idct(x, type=2))

File: crates/fsci-fft/src/bin/test_dct.rs
use fsci_fft::{dct, dct_i, dct_iii, dct_iv, idct, FftOptions};

fn main() {
    let x = vec![1.0, 2.0, 3.0, 4.0];
    let opts = FftOptions::default();
    
    println!("dct1: {:?}", dct_i(&x, &opts).unwrap());
    println!("dct2: {:?}", dct(&x, &opts).unwrap());
    println!("dct3: {:?}", dct_iii(&x, &opts).unwrap());
    println!("dct4: {:?}", dct_iv(&x, &opts).unwrap());
    println!("idct2: {:?}", idct(&x, &opts).unwrap());
}
